index.js

```diff
@@ -1,10 +1,40 @@
-var urlParse = require('url').parse;
+var url = require("fast-url-parser");
 var querystring = require('querystring');
 var isBrowser = (typeof window !== "undefined");
 
-var getDefaults = function(){
+// the node url library is quite slow & is often a code bottleneck
+// this wraps fast-url-parser to return url-parser-like results
+function urlParse(href) {
+  var parsed = url.parse(href);
+  var result = {
+    // fields from fast-url-parser
+    _protocol: parsed._protocol,
+    _href: parsed._href,
+    _port: parsed._port,
+    _query: parsed._query,
+    auth: parsed.auth,
+    slashes: parsed.slashes,
+    host: parsed.host,
+    hostname: parsed.hostname,
+    hash: parsed.hash,
+    search: parsed.search,
+    pathname: parsed.pathname,
+    _prependSlash: parsed._prependSlash,
+
+    // fields that node url library returns too
+    port: parsed._port === -1 ? null : parsed._port.toString(),
+    path: (parsed.pathname || "") + (parsed.search || ""),
+    href: href,
```

Comment:
the one difference between this behavior & the original behavior is that `url` adds a trailing slash to `href` before search/hash